lib/src/model/model_element.dart

```diff
@@ -237,7 +189,7 @@ abstract class ModelElement extends Canonicalization
   // TODO(jcollins-g): Allow e to be null and drop extraneous null checks.
   // TODO(jcollins-g): Auto-vivify element's defining library for library
   // parameter when given a null.
-  factory ModelElement._from(
+  factory ModelElement.for_(
```

Comment:
Also, what is the ending `_` in the name for? Why not just `ModelElement.for`?

Reply:
This is because `for` is a reserved keyword. 🤷 

I'm open to suggestions, but I would like to stay away from `from`, because of the reasons in the summary. `of` is not great either, I think.

Reply:
Ah, I forgot about that. `for_` is fine to me.